drivers/can/Kconfig.stm32fd

```diff
@@ -0,0 +1,40 @@
+# STM32 CAN configuration options
+
+DT_COMPAT_STM32_FDCAN := st,stm32-fdcan
+
+config CAN_STM32FD
+	bool
+	default $(dt_compat_enabled,$(DT_COMPAT_STM32_FDCAN))
```

Comment:
It makes sense to automatically enable the FD driver based on the devicetree node. Ideally, the CAN driver `CAN_STM32` for older MCUs would be disabled automatically aswell (in `Kconfig.stm32`) as it still shows up in menuconfig and causes compile errors if enabled.